buf.yaml

```diff
@@ -0,0 +1,27 @@
+# buf.yaml files should always be in the root of your repository.
+build:
+  roots:
+    # roots is the directories that all .proto files are contained within, and
+    # all imports must derive from.
+    # Only .proto files within these directory will be compiled.
+    - model
+  # excludes is the list of directories within root to exclude.
+  excludes:
+    - model/build
+
+# lint contains the options for lint checks.
+lint:
+  # use is the list of checker categories and ids to use for buf check lint.
+  use:
+    # The BASIC category includes everything from the MINIMAL category, as well as the STYLE_BASIC category.
+    # The MINIMAL category represents what we consider to be fundamental rules for modern Protobuf development,
+    # regardless of style
+    - BASIC
+    # checks that filenames are lower_snake_case.
+    - FILE_LOWER_SNAKE_CASE
+  # This removes ids or categories from the use list.
+  except:
+    # checks that enums do not have the allow_alias option set.
+    - ENUM_NO_ALLOW_ALIAS
+    # checks that all files with are in a directory that matches their package name.
+    - PACKAGE_DIRECTORY_MATCH
```

Comment:
Any reason why we are removing these from the linter? @BenHenning what do you think, should we add these checks as well?

Reply:
`ENUM_NO_ALLOW_ALIAS` - if we don't add this, lint will ask us to modify the enum names with some prefix and suffix. Example - second screenshot in [this](https://github.com/oppia/oppia-android/issues/971#issuecomment-635103364) comment

Reply:
`PACKAGE_DIRECTORY_MATCH` with this it will give us `Files with package "model" must be within a directory "model" relative to root but were in directory "src/main/proto".
`

Reply:
I think we can change the analytics proto to match the expectations. I do like the idea of enforcing proper prefixes and suffixes to our enums to keep consistency. Bit of context about the analytics proto, that's part of the analytics project that @Sarthak2601 is working on. As of now no code has gone in that uses this enum so now's a good time to match it to the specifications
As far as the exploration proto goes, I will defer to @rt4914 and @BenHenning . If you'll are okay with us making these changes to the enum style then let's go for it and enforce enum style as well.
About PACKAGE_DIRECTORY_MATCH, I will defer to @BenHenning on this. I think we may be able to get the files in the right directory once we move to Bazel.

Reply:
The directory match one sounds good. I don't have a strong opinion about the enum decision, though given that Android Studio tends to encourage importing enums directly even when they're nested, it might be nice to enforce the prefix being in place. I lean toward @vinitamurthi's position here.

Reply:
Done, Excluding  PACKAGE_DIRECTORY_MATCH, PACKAGE_VERSION_SUFFIX and ENUM_VALUE_PREFIX

Reply:
Resolving this on behalf of @vinitamurthi